application.cpp

```diff
@@ -64,8 +66,19 @@ bfs::path application::get_logging_conf() const {
 
 void application::startup() {
    try {
-      for (auto plugin : initialized_plugins)
-         plugin->startup();
+      auto ioc = io_serv;
+      for (auto plugin : initialized_plugins) {
+         // order of execution for single-threaded io_service is order of post
```

Comment:
What are the odds that one day the io_service will not be single-threaded?  Could we write this job in a way that it tail-posts the next job instead of posting all jobs and relying on something that may drift in the future?  That way the safety is guaranteed by the implementation of this method and not the environment.

Reply:
There is actually no reason not to just post the initialized_plugins startup loop itself. Updated PR.

Reply:
As per our discussion. Moved setup of sig handler thread to `startup()` and move plugin startup calls back to caller thread (main thread).